### ISSUE TITLE
subscribe block lists view

### ISSUE THREAD

**Issue by thieflord06** [ClearskyApp06/ClearskyUI#29]
add subscribe block lists UI in block tabs or maybe it will make sense to make a new tab?

APIs:

blocking:
/api/v1/anon/subscribe-blocks-blocklist/{client_identifier} | list of lists that the user is blocking
/api/v1/anon/subscribe-blocks-blocklist/{client_identifier}/{nt:page}
/api/v1/anon/subscribe-blocks-blocklists/total/{client_identifier}

blocked by:
/api/v1/anon/subscribe-blocks-single-blocklist/{client_identifier} | List of lists that other users are using to block this user
/api/v1/anon/subscribe-blocks-single-blocklist/{client_identifier}/{int:page}
/api/v1/anon/subscribe-blocks-single-blocklist/total/{client_identifier} | total users blocking this user based on lists
/api/v1/anon/subscribe-blocks-single-blocklist/users/{uri} | The list of users blocking the user using the specified list

all paginates except total

**Comment by Macdonny** [comment]
@thieflord06 I can work on this if no one is assigned to it.

**Comment by Macdonny** [comment]
> add subscribe block lists UI in block tabs or maybe it will make sense to make a new tab?
> 
> APIs:
> 
> blocking: /api/v1/anon/subscribe-blocks-blocklist/{client_identifier} /api/v1/anon/subscribe-blocks-blocklist/{client_identifier}/{nt:page}
> 
> blocked by: /api/v1/anon/subscribe-blocks-single-blocklist/{client_identifier} /api/v1/anon/subscribe-blocks-single-blocklist/{client_identifier}/{int:page}

Do you know if these endpoints are working in the staging API that is currently deployed? I'm getting 500 & 503. @thieflord06 

endpoint: /api/v1/anon/subscribe-blocks-blocklist/{client_identifier}

![image](https://github.com/user-attachments/assets/0224f218-38be-4bba-9a53-d571a28831dd)

endpoint: /api/v1/anon/subscribe-blocks-single-blocklist/{client_identifier}

![image](https://github.com/user-attachments/assets/c41b49ae-7459-46ec-b985-f5f0b77d2bc4)

**Comment by thieflord06** [comment]
Ok. I'll look into it.

**Comment by thieflord06** [comment]
i found a couple of issues. I'm almost done fixing them however, when the fix is pushed, it will also push the update that will break the UI from #229

**Comment by thieflord06** [comment]
Alright, I'm pushing the update to staging backend.

**Comment by thieflord06** [comment]
I added the total endpoints in the original post above.

**Comment by Macdonny** [comment]
> I added the total endpoints in the original post above.

Thanks for the fix!

**Comment by Macdonny** [comment]
> I added the total endpoints in the original post above.

Hey, it looks like there might be a bug with the "subscribe-blocks-blocklist" endpoint. The endpoint is returning duplicates of the same list. This is the URL I'm making the call to: https://staging.api.clearsky.services/api/v1/anon/subscribe-blocks-blocklist/aoc.bsky.social

https://github.com/user-attachments/assets/ccbdd521-da6a-424f-8031-b2600fe8b554

Also is it possible to return the lists with the same format and fields as the "/api/v1/anon/get-list/" endpoint?
```javascript
  "lists": [
      {
          "created_date": "2024-11-23T05:49:17.810000+00:00",
          "date_added": "2024-11-23T05:49:17.949000+00:00",
          "description": null,
          "did": "did:plc:222bodydxxztbrzgdicdluj3",
          "name": "Tom's Starter Pack",
          "url": "https://bsky.app/profile/did:plc:222bodydxxztbrzgdicdluj3/lists/3lblsbqyzje2a"
      },
      {
         ....
      }
  ]
```

**Comment by thieflord06** [comment]
So that endpoints is returning a list of:

Users (subject_did) that are being blocked by the person (did). It gives the list (URI/URL) and the list owner (list_owner) and the date they blocked them using that list. 

So, it may look like it's the same info but the thing that's changing is the "did", which is the actual people that are being blocked. 

For clarity/better formatting, I guess I can change it to where it's a dictionary of a dictionary of the list of users blocked by that list so you only see the list info once vs every time.

**Comment by thieflord06** [comment]
After thinking about it a little, we can do the list approach. We can hyperlink the list to the list in bsky short term, and long term hyperlink it to a page with all the users on that list.

**Comment by thieflord06** [comment]
For this to work I'll have to make another endpoint for subscribe-blocks-single-blocklist to click thru that gets who is subscribed to that list vs just showing who is on the list. 
(in progress)

**Comment by thieflord06** [comment]
I just added the last endpoint, ready for testing.

**Comment by thieflord06** [comment]
@Macdonny Any updates on this?

**Comment by Macdonny** [comment]
Took a pause on this, but picking this back up today.

**Comment by thieflord06** [comment]
@Macdonny any progress on this? This is very high on the priority list.

**Comment by Macdonny** [comment]
> [@Macdonny](https://github.com/Macdonny) any progress on this? This is very high on the priority list.

Yes, I made progress. I should have a PR tonight!

**Comment by Macdonny** [comment]
Hey @thieflord06, wrapping up this issue, but there are a few things that need to be updated on the API:

- Both of the total endpoints are returning a "503 Service Unavailable" error with this message {"error":"API is disabled"}
- Can you add the list description in both block list endpoints?
- The subscribe-blocks-single-blocklist" endpoint is returning duplicate results similar to the other endpoint:

https://github.com/user-attachments/assets/a1792b02-1ecf-4761-a129-888eefebd30b

**Comment by thieflord06** [comment]
I just activated the endpoints.

**Comment by thieflord06** [comment]
I can add the description to "who am I subscribed to" but the "who is subscribe" is only returning the users.

**Comment by thieflord06** [comment]
I can add the URL of the list easily and then we can make a separate request to get the description info for each person?

**Comment by thieflord06** [comment]
I just realized the updated workflow. To confirm for "who I am blocked by" we will have a list of lists and then that clicks into the list of users?

**Comment by Macdonny** [comment]
> I just realized the updated workflow. To confirm for "who I am blocked by" we will have a list of lists and then that clicks into the list of users (on our side)?
> 
> If so, we would have to add a "see users" maybe ro a drop down, because the traditional UX is when you click the list name hyperlink, it takes you to bsky.
> 
> the description is returning for both now.

I see the description in the data from "/subscribe-blocks-single-blocklist/"  but not the data from "/subscribe-blocks-blocklist/".

Yeah, that should be an easy add to the blocked-by lists. Also, to get the list of users, I would use the endpoint: "/subscribe-blocks-single-blocklist/users/{uri}"? Could you send an example of how I would add the URI?

**Comment by thieflord06** [comment]
Description added.

I'm actually thinking of removing the URI returns because the UI doesn't need to use it so you can pass back the URL instead.

Ex: https://staging.api.clearsky.services/api/v1/anon/subscribe-blocks-single-blocklist/users/https://bsky.app/profile/did:plc:wyhoxn2no4mjilvinxlm7ouh/lists/3l34o5euutz2m

I'm finishing up on changing over from URI to URL.

**Comment by thieflord06** [comment]
deployed.

**Comment by thieflord06** [comment]
Can you push a prelim PR so I can do testing for duplicates?

**Comment by Macdonny** [comment]
Yeah, I can. Going to push now.

**Comment by Macdonny** [comment]
@thieflord06 created the PR: https://github.com/ClearskyApp06/ClearskyUI/pull/290
